src/Temporalio/Common/SimplePluginOptions.cs

```diff
@@ -0,0 +1,245 @@
+        /// </summary>
+        /// <param name="type">Type to get activities from.</param>
+        /// <param name="instance">Instance to use when invoking. This must be non-null if any
+        /// activities are non-static.</param>
+        /// <returns>This options instance for chaining.</returns>
+        public SimplePluginOptions AddAllActivities(Type type, object? instance)
+        {
+            foreach (var defn in ActivityDefinition.CreateAll(type, instance))
+            {
+                AddActivity(defn);
+            }
+            return this;
+        }
+
+        /// <summary>
+        /// Add the given type as a workflow.
+        /// </summary>
+        /// <typeparam name="T">Type to add.</typeparam>
+        /// <returns>This options instance for chaining.</returns>
+        public SimplePluginOptions AddWorkflow<T>() => AddWorkflow(typeof(T));
+
+        /// <summary>
+        /// Add the given type as a workflow.
+        /// </summary>
+        /// <param name="type">Type to add.</param>
+        /// <returns>This options instance for chaining.</returns>
+        public SimplePluginOptions AddWorkflow(Type type) =>
+            AddWorkflow(WorkflowDefinition.Create(type));
+
+        /// <summary>
+        /// Add the given workflow definition. Most users will use <see cref="AddWorkflow{T}" />
+        /// instead.
+        /// </summary>
+        /// <param name="definition">Definition to add.</param>
+        /// <returns>This options instance for chaining.</returns>
+        public SimplePluginOptions AddWorkflow(WorkflowDefinition definition)
+        {
+            Workflows.Add(definition);
+            return this;
+        }
+
+        /// <summary>
+        /// Add the given Nexus service handler.
+        /// </summary>
+        /// <param name="serviceHandler">Service handler to add. It is expected to be an instance of
+        /// a class with a <see cref="NexusServiceHandlerAttribute"/> attribute.</param>
+        /// <returns>This options instance for chaining.</returns>
+        /// <remarks>WARNING: Nexus support is experimental.</remarks>
+        public SimplePluginOptions AddNexusService(object serviceHandler)
+        {
+            NexusServices.Add(ServiceHandlerInstance.FromInstance(serviceHandler));
+            return this;
+        }
+
+        /// <summary>
+        /// Create a shallow copy of these options.
+        /// </summary>
+        /// <returns>A shallow copy of these options and any transitive options fields.
+        /// Also copies collections of activities and workflows.</returns>
+        public virtual object Clone()
+        {
+            return (SimplePluginOptions)MemberwiseClone();
+        }
+
+        /// <summary>
+        /// Represents a required configurable plugin option.
+        /// </summary>
+        /// <typeparam name="T">The type of the option value.</typeparam>
+        public class SimplePluginOption<T>
```

Comment:
Small bump of https://github.com/temporalio/sdk-dotnet/pull/548#discussion_r2466847847. I personally think `SimplePluginOptions.Option` is clearer than `SimplePluginOptions.SimplePluginOption`, but not a big deal. Just want to make sure you saw that comment/suggestion.